File: server/schemas/translate.py
from pydantic import BaseModel


class TranslateTeamPostScheme(BaseModel):
    description: str | None
    title: str


class TranslateTeamMemberScheme(BaseModel):
    uuid: str
    team_id: int
    accepted: bool


class TranslateTeamScheme(BaseModel):
    id: int
    owner_uuid: str
    title: str
    description: str | None
    approved: bool
    accepted_members: list[TranslateTeamMemberScheme]
    unaccepted_members: list[TranslateTeamMemberScheme]

File: client/types/translate.ts
export interface TranslateTeamPostScheme {
  description: string | null;
  title: string;
}

export interface TranslateTeamMemberScheme {
  uuid: string;
  team_id: number;
  accepted: boolean;
}

export interface TranslateTeamScheme {
  id: number;
  owner_uuid: string;
  title: string;
  description: string | null;
  approved: boolean;
  accepted_members: TranslateTeamMemberScheme[];
  unaccepted_members: TranslateTeamMemberScheme[];
}
